provider/pkg/provider/diff.go

```diff
@@ -24,7 +24,9 @@ func (k *kubeProvider) forceNewProperties(obj *unstructured.Unstructured) []stri
 	props := metadataForceNewProperties(".metadata")
 	if group, groupExists := forceNew[gvk.Group]; groupExists {
 		if version, versionExists := group[gvk.Version]; versionExists {
-			if kindFields, kindExists := version[gvk.Kind]; kindExists {
+			if clients.IsSecret(obj) && !k.enableSecretMutable {
+				props = append(props, properties{".type", ".stringData", ".data"}...)
+			} else if kindFields, kindExists := version[gvk.Kind]; kindExists {
```

Comment:
@MaienM it just occurred to me while adding some detail to https://github.com/pulumi/pulumi-kubernetes/issues/3179 that we should probably respect the `immutable` field.

In other words, if the secret has `immutable: true` it should preserve the existing behavior even if the provider has `enableSecretMutable`. Does that make sense?

Reply:
That makes sense, yeah, it's not like trying to treat it as mutable will even work in that case. The same applies to `ConfigMap`s though, right? I don't think those handle that either right now, that's what I based this code on after all.

I can make a change to handle that, should I also make the same change for `ConfigMap`s while I'm at it?

Reply:
I've made the change for this for both `Secret`s and `ConfigMap`s. I put the change for the `ConfigMap`s in a separate commit because I wasn't sure whether you wanted that one, so you can just get rid of that commit if you don't.

Reply:
@MaienM thank you again for the great contribution! I'm very happy this prompted us to uncover the `immutable` incompatibility and that you were able to fix it while you were in here. I added an E2E test case partly as a sanity check and party to cement the behavior we expect here.

So much has changed with Pulumi and k8s since this behavior was originally introduced that it would make much more sense IMO to make this the default behavior at some point in the future (#3179). I added a note to the changelog to that effect, since it's a potentially disruptive behavioral change.

After I merge this it will be available as a pre-release if you'd like to play around with it.